crates/core_arch/src/x86/avx512f.rs

```diff
@@ -38007,6 +38127,34 @@ extern "C" {
     #[link_name = "llvm.x86.avx512.mask.compress.pd.128"]
     fn vcompresspd128(a: f64x2, src: f64x2, mask: u8) -> f64x2;
 
+    #[link_name = "llvm.x86.avx512.mask.compress.store.d.512"]
+    fn vcompressd_mem(mem: *mut i8, data: i32x16, mask: u16);
```

Comment:
Is there a better naming convention for these intrinsics? The asm mnemonic is the same for mem and reg operations.

Reply:
The recommended way to figure this out is to look at what IR clang generates: https://godbolt.org/z/nvaxM4MGh

In this case it is calling the `llvm.masked.compressstore.v2f64` intrinsic which unfortunately can't be called directly from Rust because it uses a `i1` vector which can't be represented with Rust types.

This is the reason why #1254 implemented some of the AVX512 intrinsics using inline assembly instead. I think this is the right approach in this case as well.

Reply:
These llvm intrinsic seem to work though, I saw them used with plain integer masks in llvm testcases: https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/X86/avx512-intrinsics-upgrade.ll#L8938 (not sure if there is an official documentation for them though). I wanted to avoid asm unless absolutely necessary.

Test failures in CI seem unrelated to the changes in this PR.